forum/profiles/models.py

```diff
@@ -111,3 +114,39 @@ def __str__(self):
             f"email={self.email}"
             f")"
         )
+
+
+class ViewedStratups(models.Model):
+    """
+    Model to track startup views by investors.
+
+    Fields:
+        startup (ForeignKey): Reference to the StartupProfile model representing the viewed startup.
+        investor (ForeignKey): Reference to the InvestorProfile model.
+        viewed_at (DateTimeField): The timestamp when the startup was viewed.
+    """
+    startup = models.ForeignKey(StartupProfile, on_delete=models.CASCADE)
+    investor = models.ForeignKey(InvestorProfile, on_delete=models.CASCADE, db_index=True)
+    viewed_at = models.DateTimeField(auto_now_add=True, db_index=True)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=['investor', 'startup'],
+                name='one_view_per_day',
+                condition=Q(viewed_at__date=TruncDate('viewed_at'))
+            )
+        ]
```

Comment:
The use of `Q` with `TruncDate` in the `UniqueConstraint` may not work as expected in all databases since constraints are implemented at the database level, and not all databases support such expressions in constraints. You should handle this logic in the application code or consider using a unique constraint on `investor`, `startup`, and `viewed_at` truncated to the day in the database.
_Alternative Approach:_
```
constraints = [
    models.UniqueConstraint(
        fields=['investor', 'startup', 'viewed_at'],
        name='one_view_per_day'
    )
]
```
Then, ensure you truncate `viewed_at` to the date level in your application logic before saving.